app/validation/validator.py

```diff
@@ -1491,11 +1520,19 @@ def _get_answers_with_parent_ids(self, json_to_validate):
         answers = {}
         for question, context in self._get_questions_with_context(json_to_validate):
             for answer in question.get('answers', []):
-                answers[answer['id']] = {
-                    'answer': answer,
-                    **context
-                }
+
+                if not answer['id'] in answers:
```

Comment:
I think this check may be redundant? Won't the context will be the same in cases where the answer id is the same?

Reply:
It checks for all possible variants of proxy and non proxy answers(which have same context but different values and labels, paired with different routing rules). If you remove that only answers from last variant will be added and validator will fail.

Reply:
Won't the current approach allow invalid answers from alternative variants then?

Reply:
Removing answer variants from pool of answers will remove some of values referenced by routing values. Routing rules are sometimes looking for specific variants of same questions, but if we add all variants of given question, we will end up with same labels and values which will break validation. So it's not all variants, it's only all different variants. Routing values are searching for answer values not the other way round. There was a bug in answer_ids_with group_ids creation when last variant of question was overwriting the previous variant, so even if you had four, you ended up with last.